### DISCUSSION TITLE
Help Simple Slave Read with Data Convert

### DISCUSSION THREAD

**Discussion by sallos** [eModbus/eModbus#297] [General]
Hi, 
i used this Code to Poll a Master:
I wont to Read a Long(int32) Value ~ -2000 to 2000
`MB.addRequest((uint32_t)previousMillis,` 1, READ_HOLD_REGISTER, 13009, 2);`

And this is the Handler:
`void handleData(ModbusMessage response, uint32_t token) 
{
  int32_t addr, words;
  response.get(2, addr);
  response.get(4, words);

  Serial.println("Watt");
  Serial.println(addr);
  Serial.println(words);
}`

But the are no possible Outputs. Is the are Sample? How can i convert the Bytes to a Long.

**Comment by Miq1** [comment]
Have a look at the description of the ``get()`` method at [the documentation pages](https://emodbus.github.io/reading-from-a-modbusmessage). As ``int32`` is an integral C++ data type, ``get()`` will handle it.

**Comment by sallos** [comment]
OK,

is it Possible so Swap the Register?
`  int32_t addr;
response.get(3, addr, SWAP_REGISTERS);`

**Reply by Miq1** [reply]
Yes, if you need that. It will convert a sequence byte1, byte2, byte3, byte4 into byte3, byte4, byte1, byte2 in your ``addr`` variable.

**Comment by sallos** [comment]
Hi,
ok this work for Values from in positive range.

But the long Value in Modbus can be also in negative Range.

Sampe Range -2000 to 2000

Modbus Message Vale = 10
01 03 04 00 0A 00 00
Value 2000
01 03 04 07 D0 00 00
Value -10
01 03 04 FF F6 FF FF
Value -2000
01 03 04 F8 30 FF FF

The Server is a Sungrow Solar and the Datatype Called  int32sw

**Reply by Miq1** [reply]
Did you use an ``int`` variable?

**Reply by Miq1** [reply]
Wait a sec... the swap parameters only will work with ``float`` values - sorry. So you will have to do the conversion by hand, like
```
int Val;
uint16_t half1Val, half2Val;
response.get(3, half2Val, half1Val);
Val = int((half1Val << 16) | half2Val);
```
(did not test that, though)
